File: Make_Array_Consecutive_2/MakeArrayConsecutive2.java
import java.util.Set;
import java.util.Arrays;
import java.util.stream.Collectors;
public class MakeArrayConsecutive2{
	public static void main (String args[]){
		int[] test = {0, 9, 6, 6, 5, 5, 1}; 
		int[] test1 = {0};

		System.out.format("You need %d more statues to have all consecutive sizes from %s%n", solution(test), Arrays.toString(test));
		System.out.format("You need %d more statues to have all consecutive sizes from %s%n", solution(test1), Arrays.toString(test1));
	}
	static int solution(int[] statues){
		Set<Integer> sizes = Arrays.stream(statues).boxed().collect(Collectors.toSet());
		int max = sizes.stream().max(Integer::compareTo).get(),
			min = sizes.stream().min(Integer::compareTo).get();
		return (max - min + 1) - sizes.size();
	}
}

File: Make_Array_Consecutive_2/Make_array_consecutive_2.js
function solution(statues){
	let neededStatues = 0
	for(var i = 1, max= statues[0], min = statues[0]; i<statues.length; i++){
		let current = statues[i];
		if(statues.indexOf(current)!=i)
			continue;

		if(current > max){
			neededStatues += current - max - 1
			max = current
		}else if(current < min){
			neededStatues += current - min -1
                        min = current
		}else {
			neededStatues--
		}
	}
	return neededStatues
}

let test = [0, 9, 6, 6, 5, 5, 1]
let test1 = [1]

console.log(`You need ${solution(test)} more statues to have all sizes from [${test}]`)
console.log("You need %d more statues to have all sizes from %s", solution(test1), test1)
